src/GoTrueClient.ts

```diff
@@ -941,6 +941,14 @@ export default class GoTrueClient {
 
     this.stateChangeEmitters.set(id, subscription)
 
+    this.getSession()
```

Comment:
Can we use async/await and log the error?

Reply:
don't think we can do that because `onAuthStateChange` is a synchronous method

Reply:
async/await is syntax sugar anyway. We can log the error if you like, but tbh the result of an error is a logged-out user, so this may not be the place for a log.

Reply:
Well that's if the error is not caused by the library. If it's like a `TypeError` this will just fail silently. I'd still insist on the logging so we're not wrecking our brains. Regarding async/await, you can refactor this into a new async method like `emitInitialSession()` which is async but is not being awaited in onAuthStateChange.

Reply:
> Regarding async/await, you can refactor this into a new async method like emitInitialSession() which is async but is not being awaited in onAuthStateChange.

@hf hmm can you clarify what you mean by this? `getSession` is already an async method and it has proper error handling in it. Or are you suggesting we create an async version of `onAuthStateChange` instead?

Reply:
You could also accept an arg on the `.catch()` callback and log that

Reply:
```ts
      .catch((error) => {
         console.error(error)
         this.stateChangeEmitters.get(id)?.callback('INITIAL_SESSION', null)
       })
```